perf/smallbank_workload/src/smallbank_tranformer.rs

```diff
@@ -80,3 +143,30 @@ impl<'a> SBPayloadTransformer<'a> {
         Ok(txn)
     }
 }
+
+struct DependencyTracker<T>
+where
+    T: Eq + Hash,
+{
+    signature_by_id: HashMap<T, String>,
+}
+
+impl<T> DependencyTracker<T>
+where
+    T: Eq + Hash,
+{
+    pub fn new() -> DependencyTracker<T> {
+        DependencyTracker {
+            signature_by_id: HashMap::new(),
+        }
+    }
+
+    pub fn get_dependency_for_id(&self, id: &T) -> Option<&String> {
```

Comment:
Just call this `get_dependency`

Reply:
Makes sense, since the id is in the signature.